src/transformers/trainer.py

```diff
@@ -3035,9 +3035,10 @@ def _save_tpu(self, output_dir: Optional[str] = None):
 
         # Save a trained model and configuration using `save_pretrained()`.
         # They can then be reloaded using `from_pretrained()`
+        supported_classes = (PreTrainedModel,) if not is_peft_available() else (PreTrainedModel, PeftModel)
```

Comment:
I think all `PushtoHubMixin` can fall in the classes that support save_pretrained and from pretrained so we could also use that as both should inherit from the latter

Reply:
@ArthurZucker perfect idea, let me give it a try :) and will update it.

Reply:
My fix was inspired from the code here: https://github.com/huggingface/transformers/blob/0cb946d3f2dd6c91dc9462ba3589d35e7bcbc618/src/transformers/trainer.py#L3073

Reply:
@ArthurZucker if I understood correctly, I tried the following change:

`supported_classes = (PushToHubMixin, )` in [line 3073](https://github.com/huggingface/transformers/blob/0cb946d3f2dd6c91dc9462ba3589d35e7bcbc618/src/transformers/trainer.py#L3073) but it didn't work.